mllib/src/main/scala/org/apache/spark/ml/ann/Layer.scala

```diff
@@ -208,14 +214,18 @@ private[ann] object AffineLayerModel {
    * Generate random weights for the layer
    * @param numIn number of inputs
    * @param numOut number of outputs
-   * @param seed seed
+   * @param weights vector for weights initialization
+   * @param random random number generator
    * @return (matrix A, vector b)
    */
-  def randomWeights(numIn: Int, numOut: Int, seed: Long = 11L): (BDM[Double], BDV[Double]) = {
-    val rand: XORShiftRandom = new XORShiftRandom(seed)
-    val weights = BDM.fill[Double](numOut, numIn){ (rand.nextDouble * 4.8 - 2.4) / numIn }
-    val bias = BDV.fill[Double](numOut){ (rand.nextDouble * 4.8 - 2.4) / numIn }
-    (weights, bias)
+  def randomWeights(numIn: Int, numOut: Int, weights: BDV[Double], random: Random):
+  (BDM[Double], BDV[Double]) = {
```

Comment:
nit: I would prefer to line break at arguments rather than return type. Not sure if it's part of official style guide but it's what I've seen elsewhere